ocaml/networkd/bin/network_server.ml

```diff
@@ -1474,10 +1474,21 @@ end
 module PVS_proxy = struct
   open S.PVS_proxy
 
-  let path = ref "/run/pvsproxy"
+  let path = ref ""
+
+  let depriv_path = "/run/pvsproxy-state/socket"
+
+  let legacy_path = "/opt/citrix/pvsproxy/socket/pvsproxy"
+
+  let default_path () =
+    if Sys.file_exists depriv_path then
+      depriv_path
+    else
+      legacy_path
 
   let do_call call =
-    try Jsonrpc_client.with_rpc ~path:!path ~call ()
+    let p = match !path with "" -> default_path () | path -> path in
```

Comment:
Do we need to check each time to cope with the update case? Otherwise we can update `path` to cache the result.

Reply:
I had cached originally but it doesn't work if pvsproxy gets updated when networkd is running. Given that these calls are relatively infrequent and this code can be removed in the not-too-distant future, I considered it a reasonable compromise to check every time.